### ISSUE TITLE
Example for `Lrnr_cv_selector` is not working

### ISSUE THREAD

**Issue by emmavestesson** [jeremyrcoyle/sl3#401]
Hi, 

The example for Lrnr_cv_selector does not seem to work - it prints the final function instead of evaluating it (see code copied from example below). I think the final line is missing brackets with the eval_function option specified (based on  [section 6.5.1 in the tlverse handbook](https://tlverse.org/tlverse-handbook/sl3.html?#including-ensemble-super-learners-as-candidates-in-discrete-super-learner)). Can I also ask what loss function to specify for a binary outcome? Am I right to think that `loss_squared_error` is for a continuous outcome?

```r
[data](https://rdrr.io/r/utils/data.html)(cpp_imputed)
covs <- [c](https://rdrr.io/r/base/c.html)("apgar1", "apgar5", "parity", "gagebrth", "mage", "meducyrs")
task <- [sl3_Task](https://tlverse.org/sl3/reference/sl3_Task.html)$new(cpp_imputed, covariates = covs, outcome = "haz")

hal_lrnr <- [Lrnr_hal9001](https://tlverse.org/sl3/reference/Lrnr_hal9001.html)$new(
  max_degree = 1, num_knots = [c](https://rdrr.io/r/base/c.html)(20, 10), smoothness_orders = 0
)
lasso_lrnr <- [Lrnr_glmnet](https://tlverse.org/sl3/reference/Lrnr_glmnet.html)$new()
glm_lrnr <- [Lrnr_glm](https://tlverse.org/sl3/reference/Lrnr_glm.html)$new()
ranger_lrnr <- [Lrnr_ranger](https://tlverse.org/sl3/reference/Lrnr_ranger.html)$new()
lrnrs <- [c](https://rdrr.io/r/base/c.html)(hal_lrnr, lasso_lrnr, glm_lrnr, ranger_lrnr)
[names](https://rdrr.io/r/base/names.html)(lrnrs) <- [c](https://rdrr.io/r/base/c.html)("hal", "lasso", "glm", "ranger")
lrnr_stack <- [make_learner](https://tlverse.org/sl3/reference/Lrnr_base.html)(Stack, lrnrs)
metalrnr_discrete_MSE <- Lrnr_cv_selector$new(loss_squared_error)
discrete_sl <- [Lrnr_sl](https://tlverse.org/sl3/reference/Lrnr_sl.html)$new(
  learners = lrnr_stack, metalearner = metalrnr_discrete_MSE
)
discrete_sl_fit <- discrete_sl$train(task)
discrete_sl_fit$cv_risk
```

**Comment by rachaelvp** [comment]
Hi @emmavestesson,

Thanks for filing an issue. You're right that the final line in your code (`discrete_sl_fit$cv_risk`) prints the final function instead of evaluating it, and this is because `cv_risk` is a function. (In general, to evaluate any function in R, the function name needs to be followed by parenthesis, with any required arguments it supplied within the parenthesis.) In the final line of your example, if you replace `discrete_sl_fit$cv_risk` with `discrete_sl_fit$cv_risk(eval_function = loss_squared_error)`, then the CV risk will be evaluated and returned. There's an example of this formatting the handbook, here: https://tlverse.org/tlverse-handbook/sl3.html?#cross-validated-predictive-performance. I don't see where the error is in [section 6.5.1 in the tlverse handbook](https://tlverse.org/tlverse-handbook/sl3.html?#including-ensemble-super-learners-as-candidates-in-discrete-super-learner)?

Regarding which loss function to choose for prediction of a binary outcome, what is your goal? Regression or classification? For regression (e.g., if you're interested in estimating the risk of an event), I recommend negative log-likelihood loss, and you can specify this with `loss_loglik_binomial` in `sl3`. For classification, the choice of evaluation can be defined by the metric that you care about optimizing (e.g., AUC). If your desired metric to optimize for binary classification is available as a performance measure in the `ROCR` R package, then you can use the `custom_ROCR_risk` factory function in `sl3` to define a custom risk for a given `ROCR` measure. (A complete list of available performance measures in the `ROCR` R package is listed on page 3 of that package's vignette: https://cran.r-project.org/web/packages/ROCR/ROCR.pdf.)

**Comment by emmavestesson** [comment]
Hi,
 
The error is not in the handbook but in the help file in the `sl3` package for the Lrnr_cv_selector function (so when running ?Lrnr_cv_selector). Sorry, when rereading my question I realise that was not super clear. 

Thanks for clarifying the loss functions. I have read your paper on specifying a super learner and found it really useful but I wasn't sure what to change the loss function to and I couldn't find the R code that you mention in the paper.

**Comment by rachaelvp** [comment]
Hi @emmavestesson, I see. I am about to submit a PR to the devel branch that contains an update to the help file for `Lrnr_cv_selector`. Thanks for finding this! 

So happy you found that paper useful. The R code is in the supplement and that will be available in the peer-reviewed version of the paper. In `sl3`, loss functions are typically supplied via the argument `eval_function` (e.g., `Lrnr_cv_selector` and `cv_risk`), so you can change the loss function by supplying another loss to `eval_function`.
